llvm/lib/Target/DirectX/DXILShaderFlags.cpp

```diff
@@ -100,6 +100,28 @@ static bool checkWaveOps(Intrinsic::ID IID) {
   }
 }
 
+// Checks to see if the status bit from a load with status
+// instruction is ever extracted.
+// This is our proof that the module requires TiledResources
+// to be set, as if check access fully mapped was used.
+bool checkIfStatusIsExtracted(const Instruction &I) {
+  auto *II = dyn_cast<IntrinsicInst>(&I);
+  assert(II);
+  auto IID = II->getIntrinsicID();
+  assert(IID == Intrinsic::dx_resource_load_typedbuffer ||
+         IID == Intrinsic::dx_resource_load_rawbuffer);
```

Comment:
If you change the signature of the function to take an `const IntrinsicInst &` instead of an `Instruction` you wouldn't need the extra cast and assert here. The callers already have an `IntrinsicInst` in the form of `II` anyway!

Also please include a message in the assert - it makes things easier to figure out without necessarily having to open up the source file at the line of the assert.